src/machines/java/com/enderio/machines/common/blockentity/solar/SolarPanelEnergyStorageWrapper.java

```diff
@@ -0,0 +1,74 @@
+package com.enderio.machines.common.blockentity.solar;
+
+import com.enderio.api.io.IIOConfig;
+import com.enderio.api.io.energy.EnergyIOMode;
+import com.enderio.machines.common.io.energy.MachineEnergyStorage;
+import dev.gigaherz.graph3.Graph;
+import dev.gigaherz.graph3.GraphObject;
+import dev.gigaherz.graph3.Mergeable;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class SolarPanelEnergyStorageWrapper extends MachineEnergyStorage {
+
+    @Nullable
+    private Graph<Mergeable.Dummy> graph;
+
+    private final ISolarPanelTier tier;
+
+    public SolarPanelEnergyStorageWrapper(IIOConfig config, EnergyIOMode ioMode, Supplier<Integer> capacity, Supplier<Integer> usageRate, ISolarPanelTier tier) {
+        super(config, ioMode, capacity, usageRate);
+        this.tier = tier;
+    }
+
+    public void setGraph(Graph<Mergeable.Dummy> graph) {
+        this.graph = graph;
+    }
+
+    @Override
+    public int getEnergyStored() {
+        if (graph == null)
+            return 0;
+        int cumulativeEnergy = 0;
+        for (GraphObject<Mergeable.Dummy> object : graph.getObjects()) {
+            if (object instanceof SolarPanelNode panelNode) {
+                cumulativeEnergy += panelNode.getInternal().get().getEnergyStored();
+            }
+        }
+        return cumulativeEnergy;
+    }
+
+    @Override
+    public int getMaxEnergyStored() {
+        if (graph == null)
+            return 0;
+        return graph.getObjects().size() * tier.getStorageCapacity();
+    }
+
+
+    @Override
+    public int takeEnergy(int energy) {
+        if (graph == null)
+            return 0;
+
+        int cumulativeEnergy = 0;
+        List<GraphObject<Mergeable.Dummy>> nodes = new ArrayList<>(graph.getObjects());
+        Collections.shuffle(nodes);
```

Comment:
I'm not sure if shuffling is worth it? Sure it means some aren't as full as others; but it'd be much less expensive, especially if we're pulling from many many conduits many times per second. This could get expensive for larger nodes.

Reply:
shuffling runs in linear time with only n*2 accesses and writes with n being the amount of nodes in this graph. This becoming expensive is a non issue, considering the limitless conduit transfer rate and people not putting hundreds of connection on them. imo it's worth it, as not shuffling would decrease the combined buffer.
scenario: 10 photovoltaoic cells and you need exactly 9 for energy (and it's perma day time), non shuffling would only fill a single cell, while shuffling would "fill" all cells equally

Reply:
That seems reasonable then, cheers for the breakdown :)